app/models/mixins/miq_web_server_runner_mixin.rb

```diff
@@ -33,4 +33,22 @@ def start_rails_server(options)
       server.start
     end
   end
+
+  def do_heartbeat_work
+    log_long_running_requests
+  end
+
+  CHECK_LONG_RUNNING_REQUESTS_INTERVAL = 30.seconds
+  def log_long_running_requests
+    @last_checked_hung_requests ||= Time.now.utc
+    return if @last_checked_hung_requests > CHECK_LONG_RUNNING_REQUESTS_INTERVAL.ago
+
+    RequestStartedOnMiddleware.long_running_requests.each do |request, duration, thread|
+      message = "Long running http(s) request: '#{request}' handled by ##{Process.pid}:#{thread.object_id.to_s(16)}, running for #{duration.round(2)} seconds"
```

Comment:
I debated having the middleware method return the PID:TID format and rounded duration but chose to keep raw data there.  I can be convinced to change it. 😉

Reply:
Does seem like this would be part of rails rails itself. Or part of new relic / scout / skylight

This will be nice to proactively track stuff down.
Is there a way to make this chartable rather than dumping into another log?

Reply:
Yes, I can see making this more generic (it assumes a threaded web server like puma).

We had looked at an alternative, https://github.com/heroku/rack-timeout, which warns about the implications of raising a timeout like it does and how it's not for production... more for debugging... This PR was not meant to take action other than notify us of a long running request.  We do not care if the threads are deadlocked or just doing too many things slowly, we just log the long running request.  I honestly don't know that we ever want to take action other than log/notify people. 

So, yes, if this works out for us, I can see making this a small standalone rack middleware with a generic interface.